integreat_cms/cms/utils/content_utils.py

```diff
@@ -162,3 +163,45 @@ def fix_alt_texts(content: HtmlElement) -> None:
                 image.attrib["alt"] = media_file.alt_text
         else:
             logger.warning("Empty img tag was found.")
+
+
+def hide_anchor_tag_around_image(content: HtmlElement) -> None:
+    """
+    This function checks whether an image tag wrapped by an anchor tag has an empty alt tag, if so it hides anchor tag from screen-reader and tab-key
+
+    :param content: the content which has an anchor tag wrapped around an img tag
+    """
+
+    for anchor in content.iter("a"):
+        # Check if the anchor tag has only img children and no other text content
+        if (
+            all(child.tag == "img" for child in anchor.iterchildren())
+            and not anchor.text_content().strip()
+        ):
+            for img in anchor.iter("img"):
+                if img.attrib.get("alt", ""):
+                    if "aria-hidden" in anchor.attrib:
+                        del anchor.attrib["aria-hidden"]
+                        logger.debug(
+                            "Removed 'aria-hidden' from anchor: %r",
+                            tostring(anchor, encoding="unicode"),
+                        )
+                    if "tabindex" in anchor.attrib:
+                        del anchor.attrib["tabindex"]
+                        logger.debug(
+                            "Removed 'tabindex' from anchor: %r",
+                            tostring(anchor, encoding="unicode"),
+                        )
+                else:
+                    # Hide the anchor tag by setting aria-hidden attribute if the image alt text is empty
+                    anchor.set("aria-hidden", "true")
+                    logger.debug(
+                        "Set 'aria-hidden' to true for anchor: %r",
+                        tostring(anchor, encoding="unicode"),
+                    )
+                    # Unfocus the anchor tag from tab key
+                    anchor.set("tabindex", "-1")
+                    logger.debug(
+                        "Set 'tabindex' to -1 for anchor: %r",
+                        tostring(anchor, encoding="unicode"),
+                    )
```

Comment:
Not sure how relevant this is, but if an anchor has multiple images (and no other nodes) as children, it seems like the presence of `aria-hidden` and `tabindex=-1` is only dependent on the very last image. Can we imagine any situation where this would pose a problem, and if yes, how would the desired behaviour look?

Reply:
This is what I commented regarding that:

>There would still be a slight problem if we had a link with multiple images, where some images have alt tags and some don't, but I think this is probably not too important and also not very likely to happen.

But fixing the problem shouldn't be too much effort either :shrug:

Reply:
I think I figured out a good way to handle this: Our original issue is that for useless links to images get read out by their URL by screen readers, as there is no readable text content inside. The proposed code fixes that in a way that makes it also run for links with multiple images, but the problem it tries to address should only occur if *none* of the images have `alt` attributes. It should not occur as soon as one does have an `alt` attribute, at least if I understand the function of screen readers correctly.

```suggestion
    for anchor in content.iter("a"):
        # Check if the anchor tag has only img children and no other text content
        if (
            all(child.tag == "img" for child in anchor.iterchildren())
            and not anchor.text_content().strip()
        ):
            if all(img.attrib.get("alt", "") for img in anchor.iter("img")):
                if "aria-hidden" in anchor.attrib:
                    del anchor.attrib["aria-hidden"]
                    logger.debug(
                        "Removed 'aria-hidden' from anchor: %r",
                        tostring(anchor, encoding="unicode"),
                    )
                if "tabindex" in anchor.attrib:
                    del anchor.attrib["tabindex"]
                    logger.debug(
                        "Removed 'tabindex' from anchor: %r",
                        tostring(anchor, encoding="unicode"),
                    )
            else:
                # Hide the anchor tag by setting aria-hidden attribute if the image alt text is empty
                anchor.set("aria-hidden", "true")
                logger.debug(
                    "Set 'aria-hidden' to true for anchor: %r",
                    tostring(anchor, encoding="unicode"),
                )
                # Unfocus the anchor tag from tab key
                anchor.set("tabindex", "-1")
                logger.debug(
                    "Set 'tabindex' to -1 for anchor: %r",
                    tostring(anchor, encoding="unicode"),
                )
```

Reply:
I think it should occur even if one out of multiple images has not an empty alt tag. A screenreader might detect them as a decorative and read out its own `href`path out of context, it might be better to hide an anchor tag completely :thinking: The whole purpose of an anchor tag in this context, AFAIK, it serves as a possibility to enlarge them

Reply:
can we just say then `any ` instead of all? what do you think @PeterNerlich @david-venhoff ?

Reply:
> The whole purpose of an anchor tag in this context, AFAIK, is to enlarge them

That is where I'm completely unsure. My assumption was that the only naturally occurrence of an anchor meant for enlarging the image is when it contains a single image.

Reply:
I mean, we could start by only applying the aria attribute if the image is the only child of an anchor. If it later turns out that we have a usecase for an anchor with multiple images, we can still change the code later.

Reply:
I guess it could look like this then. Though I'm not too sure whether the excessive use of walrus operators actually makes this more readable than alternatives.

I also flipped the cases fore adding/removing `aria-hidden` and `tabindex=-1`, I think it was still the way it was meant to be when the condition was negated.

```suggestion
    for anchor in content.iter("a"):
        # Check if the anchor tag has only img children and no other text content
        if (
            len(children := [child for child in anchor.iterchildren()]) == 1
            and (img := children[0]).tag == "img"
            and not anchor.text_content().strip()
        ):
            if img.attrib.get("alt", ""):
                # Hide the anchor tag by setting aria-hidden attribute if the image alt text is empty
                anchor.set("aria-hidden", "true")
                logger.debug(
                    "Set 'aria-hidden' to true for anchor: %r",
                    tostring(anchor, encoding="unicode"),
                )
                # Unfocus the anchor tag from tab key
                anchor.set("tabindex", "-1")
                logger.debug(
                    "Set 'tabindex' to -1 for anchor: %r",
                    tostring(anchor, encoding="unicode"),
                )
            else:
                if "aria-hidden" in anchor.attrib:
                    del anchor.attrib["aria-hidden"]
                    logger.debug(
                        "Removed 'aria-hidden' from anchor: %r",
                        tostring(anchor, encoding="unicode"),
                    )
                if "tabindex" in anchor.attrib:
                    del anchor.attrib["tabindex"]
                    logger.debug(
                        "Removed 'tabindex' from anchor: %r",
                        tostring(anchor, encoding="unicode"),
                    )
```

Reply:
> I guess it could look like this then. Though I'm not too sure whether the excessive use of walrus operators actually makes this more readable than alternatives.
> 
> I also flipped the cases fore adding/removing `aria-hidden` and `tabindex=-1`, I think it was still the way it was meant to be when the condition was negated.

I thought we can leave it like this for now, but your suggestion seems to be bettter :)